src/esp32/analog.rs

```diff
@@ -19,22 +23,36 @@ where
         name: String,
         channel: AdcChannelDriver<'a, T, ATTEN>,
         driver: Rc<RefCell<AdcDriver<'a, T::Adc>>>,
+        pin: Esp32RawAdcChannel,
     ) -> Self {
         Self {
             name,
             channel,
             driver,
+            pin,
         }
     }
     fn inner_read(&mut self) -> anyhow::Result<u16> {
-        self.driver
-            .borrow_mut()
-            .read(&mut self.channel)
+        self.raw_reading()
             .map_err(|e| anyhow::anyhow!(format!("error while reading analog reader {e}")))
     }
     fn inner_name(&self) -> String {
         self.name.clone()
     }
+    fn raw_reading(&mut self) -> anyhow::Result<u16> {
+        let unit = T::Adc::unit();
+        let channel = self.pin.0;
+        if unit == adc_unit_t_ADC_UNIT_1 {
+            let measurement = unsafe { adc1_get_raw(channel) };
```

Comment:
Why doesn't `AdcDriver::read_raw` do what you need here?

Reply:
I withdraw my comment, we would need to update esp-idf-hal which will need us to update to rust 1.71 which snowballs into an image that will not fit anymore in the 4MB address space

Reply:
Ah, that's unfortunate. Maybe leave a comment in the hopes that we can revisit later?